module/VuFind/src/VuFind/Log/LoggerFactory.php

```diff
@@ -335,97 +297,123 @@ protected function addDebugWriter(Logger $logger, $debug)
                     break;
                 case 'notice':
-                    $max = Logger::WARN;  // Warning: warning conditions
-                    $min = Logger::NOTICE;// Notice: normal but significant condition
+                    $min = LogLevel::NOTICE;
+                    $max = LogLevel::WARNING;
                     break;
                 case 'error':
-                    $max = Logger::CRIT;  // Critical: critical conditions
-                    $min = Logger::ERR;   // Error: error conditions
+                    $min = LogLevel::ERROR;
+                    $max = LogLevel::CRITICAL;
                     break;
                 case 'alert':
-                    $max = Logger::EMERG; // Emergency: system is unusable
-                    $min = Logger::ALERT; // Alert: action must be taken immediately
+                    $min = LogLevel::ALERT;
+                    $max = LogLevel::EMERGENCY;
                     break;
                 default:
-                    // INVALID FILTER, so skip it. We must continue 2 levels, so we
-                    // continue the foreach loop instead of just breaking the switch.
                     continue 2;
             }
 
-            // Clone the submitted writer since we'll need a separate instance of the
-            // writer for each selected priority level.
-            $newWriter = clone $writer;
+            // Clone the submitted baseHandler since we'll need a separate instance
+            // for each selected priority level, allowing distinct verbosity settings.
+            $newHandler = clone $baseHandler;
 
-            // verbosity
-            if ($verbosity) {
-                if (method_exists($newWriter, 'setVerbosity')) {
-                    $newWriter->setVerbosity($verbosity);
+            // Apply verbosity if specified and supported by the handler.
+            // Check verbosity > 0, as 0 might mean no extra verbosity and 1 is default
+            if ($verbosity > 0) {
+                if (method_exists($newHandler, 'setVerbosity')) {
+                    $newHandler->setVerbosity($verbosity);
                 } else {
                     throw new \Exception(
-                        $newWriter::class . ' does not support verbosity.'
+                        $newHandler::class . ' does not support verbosity when a verbosity level is configured.'
                     );
                 }
             }
-
-            // filtering -- only log messages between the min and max priority levels
-            $filter1 = new \Laminas\Log\Filter\Priority($min, '<=');
-            $filter2 = new \Laminas\Log\Filter\Priority($max, '>=');
-            $newWriter->addFilter($filter1);
-            $newWriter->addFilter($filter2);
-
-            // add the writer
-            $logger->addWriter($newWriter);
+            $filterHandler = new FilterHandler($newHandler, $min, $max);
+            if (is_a($newHandler, "VuFind\Log\Handler\MailHandler")) {
```

Comment:
This might be a slightly better way of doing the same thing:

```suggestion
            if ($newHandler instanceof MailHandler) {
```